examples/blake2f.rs

```diff
@@ -0,0 +1,1459 @@
+                .iter()
+                .zip(v_vec_values[V_LEN as usize / 2..V_LEN as usize].iter())
+                .map(|(&v1, &v2)| split_xor_value(v1, v2))
+                .collect(),
+            final_split_bit_vec: output_vec_values
+                .iter()
+                .map(|&output| split_value_4bits(output as u128, BITS_NUMBER))
+                .collect(),
+        };
+        ctx.add(&blake2f_final_step, final_inputs);
+        // ba80a53f981c4d0d, 6a2797b69f12f6e9, 4c212f14685ac4b7, 4b12bb6fdbffa2d1
+        // 7d87c5392aab792d, c252d5de4533cc95, 18d38aa8dbf1925a,b92386edd4009923
+        println!("output = {:x?}", output_vec_values);
+    })
+}
+
+fn blake2f_super_circuit<F: PrimeField + Hash>() -> SuperCircuit<F, InputValues> {
+    super_circuit::<F, InputValues, _>("blake2f", |ctx| {
+        let single_config = config(SingleRowCellManager {}, SimpleStepSelectorBuilder {});
+        let (_, iv_table) =
+            ctx.sub_circuit(single_config.clone(), blake2f_iv_table, IV_LENGTH as usize);
+        let (_, bits_table) = ctx.sub_circuit(
+            single_config.clone(),
+            blake2f_4bits_table,
+            BITS_NUMBER as usize,
+        );
+        let (_, xor_4bits_table) = ctx.sub_circuit(
+            single_config,
+            blake2f_xor_4bits_table,
+            (BITS_NUMBER * BITS_NUMBER) as usize,
+        );
+
+        let maxwidth_config = config(
+            MaxWidthCellManager::new(250, true),
+            SimpleStepSelectorBuilder {},
+        );
+
+        let params = CircuitParams {
+            iv_table,
+            bits_table,
+            xor_4bits_table,
+        };
+        let (blake2f, _) = ctx.sub_circuit(maxwidth_config, blake2f_circuit, params);
+
+        ctx.mapping(move |ctx, values| {
+            ctx.map(&blake2f, values);
+        })
+    })
+}
+
+fn main() {
+    let super_circuit = blake2f_super_circuit::<Fr>();
+    let compiled = chiquitoSuperCircuit2Halo2(&super_circuit);
+
+    let values = InputValues {
+        round: 12,
+        // h[0] = hex"48c9bdf267e6096a 3ba7ca8485ae67bb 2bf894fe72f36e3c f1361d5f3af54fa5";
+        // h[1] = hex"d182e6ad7f520e51 1f6c3e2b8c68059b 6bbd41fbabd9831f 79217e1319cde05b";
+        h_vec: [
+            0x6a09e667f2bdc948,
+            0xbb67ae8584caa73b,
+            0x3c6ef372fe94f82b,
+            0xa54ff53a5f1d36f1,
+            0x510e527fade682d1,
+            0x9b05688c2b3e6c1f,
+            0x1f83d9abfb41bd6b,
+            0x5be0cd19137e2179,
+        ], // 8 * 64bits
+        // m[0] = hex"6162630000000000000000000000000000000000000000000000000000000000";
+        // m[1] = hex"0000000000000000000000000000000000000000000000000000000000000000";
+        // m[2] = hex"0000000000000000000000000000000000000000000000000000000000000000";
+        // m[3] = hex"0000000000000000000000000000000000000000000000000000000000000000";
+        m_vec: [0x636261, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], // 16 * 64bits
```

Comment:
We use 2 hex to represent a byte, at m[0], 0x61 is the first byte, 0x62 is the second byte, 0x63 is the third byte, the rest are 0. So when we combine 16 bytes to one value, it will be represent to 0x636261. Because the first byte is in the lowest bit.